harmonica/tests/test_eql_harmonic.py

```diff
@@ -129,6 +130,89 @@ def test_eql_harmonic_small_data_cartesian():
     npt.assert_allclose(true, profile.scalars, rtol=0.05)
 
 
+def test_eql_harmonic_build_points():
+    """
+    Check if build_points method works as expected
+    """
+    region = (-3e3, -1e3, 5e3, 7e3)
+    # Define a set of observation points with variable elevation coordinates
+    easting, northing = vd.grid_coordinates(region=region, shape=(8, 8))
+    upward = np.arange(64, dtype=float).reshape((8, 8))
+    coordinates = (easting, northing, upward)
+
+    # Define a EQLHarmonic with relative depth
+    eql = EQLHarmonic(depth=1.5e3, depth_type="relative")
+    points = eql._build_points(coordinates)
+    # Check output
+    expected_points = (easting, northing, upward - 1.5e3)
+    npt.assert_allclose(points, expected_points)
+
+    # Define a EQLHarmonic with constant depth
+    eql = EQLHarmonic(depth=1.5e3, depth_type="constant")
+    points = eql._build_points(coordinates)
+    # Check output
+    expected_points = (easting, northing, -1.5e3 * np.ones_like(easting))
+    npt.assert_allclose(points, expected_points)
+
+    # Define a EQLHarmonic passing the old relative_depth parameter
+    with warnings.catch_warnings(record=True) as warn:
+        eql = EQLHarmonic(relative_depth=4.5e3)
+        assert len(warn) == 1
+        assert issubclass(warn[-1].category, FutureWarning)
+    npt.assert_allclose(eql.depth, 4.5e3)
+    assert eql.depth_type == "relative"
+    points = eql._build_points(coordinates)
+    # Check output
+    expected_points = (easting, northing, upward - 4.5e3)
+    npt.assert_allclose(points, expected_points)
```

Comment:
Best to split these tests into multiple ones. Use a fixture for the data and parametrize the options. Otherwise it's hard to tell which part breaks when tests fail.

Reply:
Sure!